### PR TITLE
Adjust FindMeetingQuery codes to pass added tests considering optional attendees

### PR DESCRIPTION
**Pull request by KathleenQ**
The adjustment is made to pass the added tests in FindMeetingQueryTest for optional attendees.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 1]
```diff
@@ -34,10 +34,13 @@
   // Some people that we can use in our tests.
```
Comment (KathleenQ) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: Please ignore all changes in this "FindMeetingQueryTest" file! They are already reviewed in PR #14 .

**Review by aannie-y — COMMENTED** [review]
LGTM however as Robert mentioned previously, try have each test in a new PR.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -26,12 +26,29 @@
...
+      return availableTime;
+    } else {
+      List<TimeRange> mandatoryUnavailableTimeWithOverlap =
+          getAttendeesUnavailableTime(events, mandatoryAttendees);
+      List<TimeRange> mandatoryUnavailableTime =
+          getNoOverlapTime(mandatoryUnavailableTimeWithOverlap);
+      List<TimeRange> mandatoryAvailableTime =
+          getAvailableTime(mandatoryUnavailableTime, meetingDuration);
+      return mandatoryAvailableTime;
+    }
```
Comment (tripd22) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Optional: I usually like to use plurals as the names for lists, e.g. `unavailableTimes`, `mandatoryAvailableTimes`, etc

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -26,12 +26,29 @@
...
+    List<TimeRange> availableTime = getAvailableTime(unavailableTime, meetingDuration);
+    if (!availableTime.isEmpty()) {
+      return availableTime;
+    } else {
+      List<TimeRange> mandatoryUnavailableTimeWithOverlap =
+          getAttendeesUnavailableTime(events, mandatoryAttendees);
+      List<TimeRange> mandatoryUnavailableTime =
+          getNoOverlapTime(mandatoryUnavailableTimeWithOverlap);
+      List<TimeRange> mandatoryAvailableTime =
+          getAvailableTime(mandatoryUnavailableTime, meetingDuration);
```
Comment (tripd22) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Optional: Single-use variables can often be refactored to make code more readable. E.g.
```
List<TimeRange> mandatoryUnavailableTimeWithOverlap =
          getAttendeesUnavailableTime(events, mandatoryAttendees);
List<TimeRange> mandatoryUnavailableTime =
          getNoOverlapTime(mandatoryUnavailableTimeWithOverlap);
List<TimeRange> mandatoryAvailableTime =
          getAvailableTime(mandatoryUnavailableTime, meetingDuration);
return mandatoryAvailableTime;
```
could become:
```
List<TimeRange> mandatoryUnavailableTime =
          getNoOverlapTime(getAttendeesUnavailableTime(events, mandatoryAttendees));
return getAvailableTime(mandatoryUnavailableTime, meetingDuration);
```

Also, do you ever envisage calling getAttendeesUnavailableTime() without immediately calling getNoOverlapTime() on the result? If not, I would be tempted to call getNoOverlapTime() within getAttendeesUnavailableTime() (i.e. return non-overlapped time directly). Your calling code could then look like:

```
List<TimeRange> mandatoryUnavailableTime =
          getAttendeesUnavailableTime(events, mandatoryAttendees);
return getAvailableTime(mandatoryUnavailableTime, meetingDuration);
```
or even:
```
return getAvailableTime(getAttendeesUnavailableTime(events, mandatoryAttendees), meetingDuration);
```

**Review by tripd22 — APPROVED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (KathleenQ) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yes. The advice really makes sense! I have adjusted all names of the lists.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (KathleenQ) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Yes, I have multiple single-use variables here by using helper functions. I have now adjusted them to be nested functions.

**Pull request merged**